src/controller/CHIPDeviceController.cpp

```diff
@@ -544,6 +540,9 @@ CHIP_ERROR DeviceCommissioner::Init(CommissionerInitParams params)
 {
     ReturnErrorOnFailure(DeviceController::Init(params));
 
+    params.systemState->SessionMgr()->RegisterCreationDelegate(*this);
```

Comment:
Please keep the TODO from https://github.com/project-chip/connectedhomeip/pull/11789/files#diff-30c3779ffb95f404ea48b1ed6319a810a1f5639a9b035002f11a44193b4cddbeL134.
That's still needed after this refactor.

```
// TODO Exchange Mgr needs to be able to track multiple delegates. Delegate API should be able to query for the right delegate
// to handle events.
```